### PR TITLE
Implement document history returning the revisions of an active document

### PR DESCRIPTION
**Pull request by wonrax**
## Type

- [x] Feature
- [ ] Bug fix
- [ ] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Brief description of the changes made.

### CODE REVIEW

File: packages/serverless/src/apollo/app/document.ts [hunk 1]
```diff
@@ -265,22 +271,26 @@ const documentDrop = authorizeWrapper<
...
+  return withTransaction(async (session) => {
+    const [listRevision, totalRevision] = await Document.history(
+      {
+        databaseName: args.databaseName,
+        collectionName: args.collectionName,
+        actor: ctx.userName,
+        docId: args.docId,
+      },
+      args.pagination || DEFAULT_PAGINATION,
+      session
```
Comment (SangTran-127) [inline packages/serverless/src/apollo/app/document.ts:284]:
```suggestion
>(JOI_DOCUMENT_HISTORY_FIND_REQUEST, async (_root: unknown,{ databaseName, collectionName, docId, pagination}, ctx) => {
  return withTransaction(async (session) => {
    const [listRevision, totalRevision] = await Document.history(
      {
        databaseName,
        collectionName,
        docId,
        actor: ctx.userName,
      },
     pagination || DEFAULT_PAGINATION,
      session
```

File: packages/serverless/src/domain/use-case/document.ts [hunk 2]
```diff
@@ -409,4 +408,56 @@ but found ${listQualifiedDocument.length}.`
...
+
+    if (!(await PermissionSecurity.collection(permissionParam, session)).read) {
+      throw new Error(
+        `Actor '${actor}' does not have 'read' permission for collection '${collectionName}' in database '${databaseName}'.`
+      );
+    }
+
+    const imDocument = ModelDocument.getInstance(databaseName, collectionName);
+
+    const document = await imDocument.findOne({ docId, active: true });
```
Comment (SangTran-127) [inline packages/serverless/src/domain/use-case/document.ts:428]: add session

File: packages/serverless/src/domain/use-case/document.ts [hunk 3]
```diff
@@ -409,4 +408,56 @@ but found ${listQualifiedDocument.length}.`
...
+          docId,
+          active: false,
+        })
+        .limit(pagination.limit)
+        .skip(pagination.offset)
+        .toArray(),
+
+      imDocument.collection.countDocuments({ docId, active: false }),
+    ]);
+
```
Comment (SangTran-127) [inline packages/serverless/src/domain/use-case/document.ts:460]: session

File: packages/serverless/src/domain/use-case/document.ts (hunk 2 above)
Comment (wonrax) [reply packages/serverless/src/domain/use-case/document.ts]: fixed

File: packages/serverless/src/domain/use-case/document.ts (hunk 3 above)
Comment (wonrax) [reply packages/serverless/src/domain/use-case/document.ts]: fixed

**Review by SangTran-127 — APPROVED** [review]

**Pull request merged**